eclair-core/src/main/scala/fr/acinq/eclair/router/Router.scala

```diff
@@ -71,7 +71,7 @@ class Router(val nodeParams: NodeParams, watcher: typed.ActorRef[ZmqWatcher.Comm
 
   val db: NetworkDb = nodeParams.db.network
 
-  {
+  private val dataHolder: VolatileRouterDataHolder = {
```

Comment:
This is overkill and will very likely create hard-to-debug performance issues. We only need two things to be accessible from the worker actors:

- the `excludedChannels`: this should be a small list, it's probably fine to copy it
- the `graphWithBalances`: this is a large data structure that we definitely don't want to copy, but we can probably make it thread-safe by simply using standard thread-safe collections? It would be useful to benchmark the cost of those changes